pkg/watch/set.go

```diff
@@ -107,6 +115,12 @@ func (w *Set) RemoveSet(other *Set) {
 }
 
 func (w *Set) Equals(other *Set) bool {
+	if w == nil && other == nil {
```

Comment:
Should these be inside the mutex? I don't know if checking for nil pointers is thread-safe

Reply:
This is safe - the method gets a copy of the pointer values as locals - the pointers themselves can't be modified by anyone else, just what is pointed at by the pointers.
Regardless, you can't dereference `w` or `other` to get at their mutex if they were nil :)

Reply:
Fair points :p